component/azstorage/connection.go

```diff
@@ -111,6 +112,8 @@ type AzConnection interface {
 
 	WriteFromFile(name string, metadata map[string]string, fi *os.File) error
 	WriteFromBuffer(name string, metadata map[string]string, data []byte) error
+	Write(name string, offset int64, len int64, data []byte, fileOffsets, modBlockList *common.BlockOffsetList) error
```

Comment:
convert all these params to structure

Reply:
I can make that change in a separate pr to change all the azstorage methods to take in options bags, I don't think this change should be made here.